tests/ashley/api/test_api_manage_moderators.py

```diff
@@ -211,13 +212,184 @@ def test_access_can_manage_moderators_list_non_moderators(self):
+        # Creates a forum
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        assign_perm("can_read_forum", user, forum, True)
+
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.patch(f"/api/v1.0/users/{user.id}/{action}/")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {"detail": "You do not have permission to perform this action."},
+        )
+
+    def test_access_api_moderator_group_instructor_post(self):
+        """Instructor can't update user with a POST to add or remove the group moderator."""
+        user = UserFactory(public_username="user")
+        # Creates a forum
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        assign_perm("can_manage_moderator", user, forum, True)
+
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.post(f"/api/v1.0/users/{user.id}/{action}/")
+        self.assertEqual(response.status_code, 405)
+        self.assertEqual(response.json(), {"detail": 'Method "POST" not allowed.'})
+
+    def test_access_api_moderator_group_instructor_put(self):
+        """Instructor can't update user with a put to add or remove the group moderator."""
+        user = UserFactory(public_username="user")
+        # Creates a forum
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        assign_perm("can_manage_moderator", user, forum, True)
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.put(f"/api/v1.0/users/{user.id}/{action}/")
+        self.assertEqual(response.status_code, 405)
+        self.assertEqual(response.json(), {"detail": 'Method "PUT" not allowed.'})
+
+    def test_access_api_can_manage_moderators_update_student_promote_revoke(self):
         """
         Promote and revoke a user with right context, permission, group
```

Comment:
not revoke, only adding the moderator group in this test.
Add another test for remove?